pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+
+        """
+        if InstanceName.namespace is None:
+            iname = InstanceName.copy()
+            iname.namespace = namespace
+        else:
+            iname = InstanceName
+
+        if iname.namespace != namespace:
+            raise CIMError(
+                CIM_ERR_INVALID_NAMESPACE,
+                _format("DeleteInstance Namespace {0!A} in InstanceName {1!A} "
+                        "does not match provided namespace.", iname.namespace,
+                        iname, namespace))
+
+        # Validate namespace and get instance CIM repository
+        instance_store = self.get_instance_store(namespace)
+
+        if not self._class_exists(namespace, iname.classname):
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} in namespace {1!A} not found. "
+                        "Cannot delete instance {2!A}",
+                        iname.classname, namespace, iname))
+
+        if not instance_store.exists(iname):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Instance {0!A} not found in CIM repository namespace "
+                        "{1!A}", iname, namespace))
+
+        # Handle namespace deletion, currently hard coded.
+        # Issue #2062 TODO/AM 8/18 Generalize the hard coded handling into
+        # provider concept
+        classname_lower = iname.classname.lower()
+        if classname_lower == 'pg_namespace':
+            ns_classname = 'PG_Namespace'
+        elif classname_lower == 'cim_namespace':
+            ns_classname = 'CIM_Namespace'
+        else:
+            ns_classname = None
+        if ns_classname:
+            namespace = iname.keybindings['Name']
+
+            # Reflect the namespace deletion in the CIM repository
+            # This will check the namespace for being empty.
+            self.remove_namespace(namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store.delete(iname)
+
+    ####################################################################
+    #
+    #   EnumerateInstances server request
+    #
+    ####################################################################
+
+    def EnumerateInstances(self, namespace, ClassName, LocalOnly=None,
+                           DeepInheritance=None, IncludeQualifiers=None,
+                           IncludeClassOrigin=None, PropertyList=None):
+        # pylint: enable:too-long
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateInstances`.
+
+        Gets a list of subclasses of `ClassName` or of the class hiearchy root
+        in the CIM repository then executes get instance for each to create
+        the list of instances to be returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+            (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
I redid all of the text for the namespace parameter to be the same so this should be fixed.

**DONE**